### ISSUE TITLE
System: Firmware: Settings: Add new option "always reboot after update"

### ISSUE THREAD

**Issue by Starkstromkonsument** [opnsense/core#6940]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Is your feature request related to a problem? Please describe.**

Some OPNsense updates do not require a reboot. Updated services are not being restarted automatically during the update procedure though. This results in continuing running the old version of the updated services, instead of the newly installed version.

Please refer to the discussion at https://github.com/opnsense/plugins/issues/3623 for further details.

**Describe the solution you like**

1. Add a warning to the WUI and Docs, that services are not being restarted automatically during the update procedure, unless it ends with a system reboot.
2. Add an option "always reboot after update" to System: Firmware: Settings to force a reboot after every update.

**Describe alternatives you considered**

1. Add the warning as stated above.
2. Add a new command "Automatic firmware update (always reboot)" to System: Settings: Cron

**Additional context**

None.

**Comment by fichtner** [comment]
> Add a warning to the WUI and Docs, that services are not being restarted automatically during the update procedure, unless it ends with a system reboot.

I don't mind adding this to the docs, but it's probably not worth the warning. The expectation of not documenting it is met by not restarting it so we're not stating anything that's not true. The biggest issue as we saw is actually getting it right, but that's not of interest to the documentation. So a sentence should be fine there.

> Add an option "always reboot after update" to System: Firmware: Settings to force a reboot after every update.

Now that we have advanced settings it could be easily anchored there indeed. But we will discuss this briefly internally.


Cheers,
Franco
